### PR TITLE
ERMS-2847 + ERMS-1989

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: app/grails-app/controllers/de/laser/PersonController.groovy [hunk 1]
```diff
@@ -61,7 +62,67 @@ class PersonController extends AbstractDebugController {
...
+                    params.list('functionType').each {
+                        PersonRole personRole
+                        RefdataValue functionType = RefdataValue.get(it)
+                        personRole = new PersonRole(prs: personInstance, functionType: functionType, org: contextOrg)
+
+                        if (PersonRole.findWhere(prs: personInstance, org:  contextOrg, functionType: functionType)) {
+                            log.debug("ignore adding PersonRole because of existing duplicate")
+                        }
+                        else if (personRole) {
+                            if (personRole.save(flush: true)) {
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/PersonController.groovy:77]: flush needed?

File: app/grails-app/controllers/de/laser/PersonController.groovy [hunk 2]
```diff
@@ -61,7 +62,67 @@ class PersonController extends AbstractDebugController {
...
+                    params.list('positionType').each {
+                        PersonRole personRole
+                        RefdataValue positionType = RefdataValue.get(it)
+                        personRole = new PersonRole(prs: personInstance, positionType: positionType, org: contextOrg)
+
+                        if (PersonRole.findWhere(prs: personInstance, org:  contextOrg, positionType: positionType)) {
+                            log.debug("ignore adding PersonRole because of existing duplicate")
+                        }
+                        else if (personRole) {
+                            if (personRole.save(flush: true)) {
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/PersonController.groovy:97]: flush needed?

File: app/grails-app/controllers/de/laser/PersonController.groovy [hunk 3]
```diff
@@ -61,7 +62,67 @@ class PersonController extends AbstractDebugController {
...
+
+                            if (RDStore.CCT_EMAIL == rdvCT) {
+                                if (!formService.validateEmailAddress(content)) {
+                                    flash.error = message(code: 'contact.create.email.error')
+                                    return
+                                }
+                            }
+
+                            Contact contact = new Contact(prs: personInstance, contentType: rdvCT, type: rdvTY, content: content)
+                            contact.save(flush: true)
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/PersonController.groovy:122]: flush needed?

File: app/grails-app/controllers/de/laser/PersonController.groovy [hunk 4]
```diff
@@ -137,67 +198,148 @@ class PersonController extends AbstractDebugController {
...
-	                render view: 'show', model: [personInstance: personInstance, userMemberships: userMemberships]
-	                return
-	            }
-	        }
-
 	        personInstance.properties = params
 
-	        if (! personInstance.save(flush: true)) {
-	            render view: 'show', model: [personInstance: personInstance, userMemberships: userMemberships]
+	        if (!personInstance.save(flush: true)) {
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/PersonController.groovy:221]: flush needed?

File: app/grails-app/controllers/de/laser/PersonController.groovy [hunk 5]
```diff
@@ -137,67 +198,148 @@ class PersonController extends AbstractDebugController {
...
+            params.list('functionType').each {
+                PersonRole personRole
+                RefdataValue functionType = RefdataValue.get(it)
+                personRole = new PersonRole(prs: personInstance, functionType: functionType, org: contextOrg)
+
+                if (PersonRole.findWhere(prs: personInstance, org:  contextOrg, functionType: functionType)) {
+                    log.debug("ignore adding PersonRole because of existing duplicate")
+                }
+                else if (personRole) {
+                    if (personRole.save(flush: true)) {
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/PersonController.groovy:238]: flush necessary?

File: app/grails-app/controllers/de/laser/PersonController.groovy [hunk 6]
```diff
@@ -137,67 +198,148 @@ class PersonController extends AbstractDebugController {
...
+            params.list('positionType').each {
+                PersonRole personRole
+                RefdataValue positionType = RefdataValue.get(it)
+                personRole = new PersonRole(prs: personInstance, positionType: positionType, org: contextOrg)
+
+                if (PersonRole.findWhere(prs: personInstance, org:  contextOrg, positionType: positionType)) {
+                    log.debug("ignore adding PersonRole because of existing duplicate")
+                }
+                else if (personRole) {
+                    if (personRole.save(flush: true)) {
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/PersonController.groovy:266]: flush necessary?

File: app/grails-app/controllers/de/laser/PersonController.groovy [hunk 7]
```diff
@@ -137,67 +198,148 @@ class PersonController extends AbstractDebugController {
...
+
+                    if (RDStore.CCT_EMAIL == rdvCT) {
+                        if (!formService.validateEmailAddress(content)) {
+                            flash.error = message(code: 'contact.create.email.error')
+                            return
+                        }
+                    }
+
+                    Contact contact = new Contact(prs: personInstance, contentType: rdvCT, type: rdvTY, content: content)
+                    contact.save(flush: true)
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/PersonController.groovy:306]: flush necessary?

File: app/grails-app/views/templates/cpa/_personFormModal.gsp [hunk 8]
```diff
@@ -1,21 +1,29 @@
...
-<laser:serviceInjection />
-<semui:modal id="personEditModal" text="${modalText ?: message(code: 'person.create_new.label')}" msgClose="${message(code: 'default.button.cancel')}" msgSave="${message(code: 'default.button.save.label')}">
-    <g:form  id="create_person" class="ui form" url="${url}" method="POST">
+<laser:serviceInjection/>
+<semui:modal id="personEditModal" text="${modalText ?: message(code: 'person.create_new.label')}" contentClass="scrolling "
+             msgClose="${message(code: 'default.button.cancel')}"
+             msgSave="${message(code: 'default.button.save.label')}">
+    <g:form id="create_person" class="ui form" url="${url}" method="POST">
+
+        <g:if test="${contextService.getOrg()}">
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/cpa/_personFormModal.gsp:8]: It it possible to access this modal without contextOrg?

File: app/grails-app/views/templates/cpa/_personFormModal.gsp [hunk 9]
```diff
@@ -57,257 +65,269 @@
...
-                        <g:textField name="contact3_content" value=""/>
-                    </div>
+                            <g:if test="${institution}">
+                                <label for="positionOrg">
+                                    <g:message code="contact.belongesTo.label"/>
+                                </label>
+                                <g:select class="ui search dropdown"
+                                          name="positionOrg"
+                                          from="${orgList}"
+                                          value="${org?.id}"
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/cpa/_personFormModal.gsp:146]: institution vs. org?

**Review by croqueGrec09 — COMMENTED** [review]
remarks

File: app/grails-app/views/templates/cpa/_personFormModal.gsp (hunk 9 above)
Comment (Moetez16) [reply app/grails-app/views/templates/cpa/_personFormModal.gsp:146]: Ist doch auskommentiert!

**Comment by Moetez16** [comment]
Done

**Review by croqueGrec09 — APPROVED** [review]
approved

**Pull request merged**